yamllint/cli.py

```diff
@@ -87,6 +88,18 @@ def github(problem, filename):
         line += problem.desc
         return line
 
+    @staticmethod
+    def json(problem, filename):
+        line = {
+            "file": filename,
+            "level": problem.level,
+            "line": problem.line,
+            "column": problem.column,
+            "desc": problem.desc,
+            "rule": problem.rule
+        }
+        return json.dumps(line)
```

Comment:
Took this suggestion. Given there are other suggestions, I will squash this commit in with the other suggestions at the end as well.

Reply:
regarding the json list structure. I am a bit confused by the suggestion, given that we would actually want `,\n` for every entry in the list except the last entry. 

What would your suggestion be for that? Given my limited python knowledge, it may require an update to the for loop to use the built-in `enumerate()` to maintain a current index and check if the current index is the last to know when we have arrived at the last element.

Something like the below:
```python
for index, problem in enumerate(problems):
  # do some formatting
```

Reply:
Ok. I have updated the PR to include the suggestions of json_pre_all and json_post_all, but I'm struggling a bit with printing the `,\n` for all except the last element given that it uses a generator and not an actual list. 

I have tried casting to a list to get a length reference and current index, but received many errors in most test cases as a result. Please provide any guidance you may have.

Reply:
Following up. The most recent revision seems to work as I have built it, but it is probably not the best implementation